### PR TITLE
chore(buffers): switch to in-memory v2 by default, remove in-memory v1

### PR DESCRIPTION
**Pull request by blt**
This change switches the default in-memory buffer to v2 version. This variant of
the in-memory buffer has better tail latencies.

REF #11006

Signed-off-by: Brian L. Troutwine <brian@troutwine.us>

EDIT from @tobz: we've also gone ahead and entirely removed in-memory v1, too.

### CODE REVIEW

**Review by tobz — APPROVED** [review]

**Comment by tobz** [comment]
Looks good to me if the numbers end up making sense. :P

**Comment by jszwedko** [comment]
Do we still want this?

**Comment by tobz** [comment]
We do.

**Comment by tobz** [comment]
I'll take a pass at fixing the unit test failures, which should just all be related to constants that need to be adjusted to compensate for differences in the two channel implementations.

**Review by lukesteensen — APPROVED** [review]

File: src/topology/test/backpressure.rs [hunk 1]
```diff
@@ -275,7 +275,7 @@ mod test_source {
             Ok(async move {
                 for i in 0.. {
                     let _result = cx.out.send(Event::from(format!("event-{}", i))).await;
-                    counter.fetch_add(1, Ordering::Relaxed);
+                    counter.fetch_add(1, Ordering::AcqRel);
```
Comment (jszwedko) [inline src/topology/test/backpressure.rs:278]: Just curious why this changed.

Comment (tobz) [reply src/topology/test/backpressure.rs:278]: I was trying to be more precise about the ordering when debugging why switching entirely to Tokio channels didn't seem to change the numbers as expected.

**Review by lukesteensen — APPROVED** [review]
I think this is worth merging. The perf loss for things like `http_datadog_filter_blackhole` is basically an indicator of a bug in the behavior of that config, since there's no sane world in which moving to a faster channel should have a significant negative impact on their performance. We were already digging into what's going on with strange topology interactions in that soak, and this change should help that investigation.

**Pull request merged**